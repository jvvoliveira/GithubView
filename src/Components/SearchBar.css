.search{
    display: inline-block;
    float: right;
}
#field{
    width: 300px;
}