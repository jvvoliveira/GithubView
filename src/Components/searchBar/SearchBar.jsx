import React, { useContext } from "react";
import { loadingContext } from "../../App";
import { Formik, Field, Form, ErrorMessage } from "formik";
import styles from "./SearchBar.module.scss";
import * as yup from "yup";
import _pesquisar from "./_pesquisar";

const validation = yup.object().shape({
  usuario: yup.string().required("Nenhum nome informado")
});

const initialValues = {
  usuario: ""
};

const Searchbar = () => {
  const {
    showLoading,
    hideLoading,
    setUsuario,
    setRepos,
    setRepos_max,
    setPage
  } = useContext(loadingContext);

  const pesquisar = usuario => {
    _pesquisar(
      showLoading,
      hideLoading,
      usuario.usuario,
      setUsuario,
      setRepos,
      setRepos_max,
      setPage
    );
  };

  return (
    <div>
      <Formik
        initialValues={initialValues}
        onSubmit={pesquisar}
        validationSchema={validation}
      >
        <Form>
          <div>
            <Field
              name="usuario"
              className={styles.field_input}
              data-testid="inputNomeUsuario"
              type="text"
              placeholder="nome do usuÃ¡rio no github"
            />
            <Field data-testid="searchButton" type="submit" value="pesquisar" />
          </div>
          <div>
            <ErrorMessage
              className={styles.error}
              component="span"
              name="usuario"
            />
          </div>
        </Form>
      </Formik>
    </div>
  );
};

export default Searchbar;
