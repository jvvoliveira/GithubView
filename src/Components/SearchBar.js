import React, { Component } from 'react';
import './SearchBar.css';


class SearchBar extends Component {

    constructor(props) {
        super(props);

        this.state = {
            usuario: ""
        }
    }

    //via arrow function não precisamos alterar o bind de contexto
    //declaramos como queremos fazer o request (default = fetch()), bom para os testes
    pesquisar = async (event, request = fetch) =>{ 
        event.preventDefault();
        let url = `https://api.github.com/users/${this.state.usuario}/repos`;
        const response = await request(url);
        const json = await response.json();
        console.log(response);
        if(response.status === 200){
            this.props.setRepos(json);
        }else{
            this.props.setRepos([]);
        }
        // .then(resposta => {
        //     return resposta.json().then(json => {
        //         this.props.setRepos(json);
        //     })
        // });
    }


    render() {
        return (
            <div className="search">
                <form onSubmit={this.pesquisar} >
                <input data-testid="input" type="text" placeholder="nome do usuário no github" onChange={(event) => {
                    let usuario = event.target.value;
                    this.setState({ usuario });
                }} />
                <input data-testid="searchButton" type="submit" value="pesquisar" onClick={this.pesquisar} />
                </form>
            </div>
        );
    }
}

export default SearchBar;