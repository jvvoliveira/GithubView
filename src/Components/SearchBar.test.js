import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from './SearchBar';
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';
import { request } from 'http';
import { promises } from 'fs';
import {
    render,
    fireEvent,
    cleanup,
    waitForElement,
    createEvent
} from '@testing-library/react';


describe("initial test SearchBar", () => {
    it('should renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SearchBar />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    // it('should take public repos by a github user', (done) => {
    //     const setRepos = jest.fn(); //mock de função
    //     const testRenderer = TestRenderer.create(
    //         <SearchBar setRepos={setRepos}></SearchBar>
    //     );

    //     let searchBar = testRenderer.getInstance();
    //     const usuario = "jvvoliveira";
    //     searchBar.setState({ usuario });
    //     expect(searchBar.state.usuario).toEqual("jvvoliveira");

    //     const event = {
    //         //propriedade que não faz nada, apenas para ser chamada no método de pesquisar e não dar undefined
    //         preventDefault: () => { }

    //     }

    //     //fazer promise dentro de promise
    //     const request = () => {
    //         return new Promise((resolve, reject) => {
    //             resolve(() => {
    //                 return {
    //                     json: () => {
    //                         return new Promise((resolve, reject) => {
    //                             resolve(() => {
    //                                 return {
    //                                     name: 'name',
    //                                     description: 'description',
    //                                     link: 'link',
    //                                     language: 'language'
    //                                 };
    //                             });
    //                         });
    //                     }
    //                 }

    //             });
    //         });
    //     };
    //     async function test() {
    //         try {
    //             await searchBar.pesquisar(event, request);
    //             expect(setRepos).toBeCalled();
    //         } catch (error) {
    //             console.log("await devolveu reject da promisse");
    //         }
    //     }
    //     test();
    // });
    it('snapshot view public repos', () => {
        const tree = renderer.create(<SearchBar></SearchBar>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should write', () => {
        const { getByText, getByTestId, container, asFragment } = render(
            <SearchBar />
        )

        const input = getByTestId('input');
        const button = getByTestId('searchButton');

        console.log(fireEvent.change(input, {target: {value: 'jvvoliveira'}}));
        console.log(fireEvent.click(button));

        // expect(input.getAttribute('value')).toHaveTextContent(/jvvoliveira/i);
        expect(input.innerHTML).toBe('jvvoliveira');
        console.log(input);
        
    });
    // it('should call pesquisar() when press button', () =>{

    //     // const { getByText, getByTestId, container, asFragment } = render(
    //     //     <SearchBar />
    //     // )
    //     // fireEvent.click(getByTestId("searchButton"));

    //     // expect(SearchBar.pesquisar).toBeCalledTimes(1);

    // });
});

