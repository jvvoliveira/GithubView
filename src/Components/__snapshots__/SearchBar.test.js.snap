// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial test SearchBar snapshot view public repos 1`] = `
<div
  className="search"
>
  <form
    onSubmit={[Function]}
  >
    <input
      data-testid="input"
      onChange={[Function]}
      placeholder="nome do usuÃ¡rio no github"
      type="text"
    />
    <input
      data-testid="searchButton"
      onClick={[Function]}
      type="submit"
      value="pesquisar"
    />
  </form>
</div>
`;
